hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java

```diff
@@ -241,7 +243,9 @@ public Hedera(@NonNull final ConstructableRegistry constructableRegistry) {
                 () -> HapiUtils.toString(version.getServicesVersion()),
                 () -> HapiUtils.toString(version.getHapiVersion()));
 
-        // Create a record builder for any genesis records that need to be created
+        // Create a records generator for any synthetic records that need to be CREATED
+        this.recordsGenerator = new SyntheticRecordsGenerator();
+        // Create a records builder for any genesis records that need to be RECORDED
         this.genesisRecordsBuilder = new GenesisRecordsConsensusHook();
```

Comment:
Should `GenesisRecordsConsensusHook` be renamed to `GenesisRecordsBuilder`? Imo as written, this is mildly confusing.

Reply:
I'm fine with that. This class acts as both a record builder and a consensus hook, but that's definitely not clear here